resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -806,11 +797,31 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  private def updateInternalState(rpId: Int, executorId: String,
+    container: Container): Unit = synchronized {
+    val containerId = container.getId
+    if (launchingExecutorContainerIds.contains(containerId)) {
+      getOrUpdateRunningExecutorForRPId(rpId).add(executorId)
+      getOrUpdateNumExecutorsStartingForRPId(rpId).decrementAndGet()
```

Comment:
getOrUpdateNumExecutorsStartingForRPId still needs to be decremented somewhere, otherwise it leaks that container and thinks its still starting